app/src/lib/shells/win32.ts

```diff
@@ -268,6 +284,35 @@ async function findCygwin(): Promise<string | null> {
   return null
 }
 
+async function findWSL(): Promise<string | null> {
+  const system32 = Path.join(
+    process.env.SystemRoot || 'C:\\Windows',
+    'System32'
+  )
+  const wslPath = Path.join(system32, 'wsl.exe')
+  const wslConfigPath = Path.join(system32, 'wslconfig.exe')
+
+  if (!(await pathExists(wslPath))) {
+    return null
+  } else if (!(await pathExists(wslConfigPath))) {
+    log.debug(
+      `[WSL] found wsl.exe, but wslconfig.exe does not exist at '${wslConfigPath}'`
+    )
+    return null
+  }
+  const exitCode = new Promise<number>(resolve => {
+    const wslDistros = spawn(wslConfigPath, ['/list'])
+    wslDistros.on('exit', resolve)
+  })
+  if ((await exitCode) !== 0) {
+    log.debug(
+      `[WSL] found wsl.exe and wslconfig.exe, but no distros are installed`
```

Comment:
:art: can we add the exit code received to this error message, to help with troubleshooting environment issues?